docs/architecture/provisioning-api.md

```diff
@@ -0,0 +1,120 @@
+# Introduction
+
+The goal of this document is to describe requirements for Provisioner component and propose an API. 
+
+## Requirements
+
+## Non-Functional
+
+The basic requirements are defined as follows:
+
+- Provisioner API is based on GraphQL. 
+- Provisioner API is asynchronous.
+
+## Functional
+
+### Runtime management
+
+The following operations must be supported:
+
+- Provisioning
+- Upgrade
+  - Kubernetes version.
+  - Kyma version.
+- Deprovisioning
+
+### Runtime Agent connection management
+
+Resetting connection between Runtime and Management Plane Services must be supported. The goal of this functionality is to force Compass Runtime Agent installed on the Runtime to establish a new connection.
+
+### Status information retrieval
+
+The following operations must be supported:
+
+- Getting status of an asynchronous operations (e.g. Runtime provisioning)
+- Getting current status of a Runtime 
+
+# API proposal
+
+## General 
+
+The basic assumptions for API design are as follows:
+
+- Cluster provisioning and Kyma installation is considered atomic operation.
+- Provisioning, deprovisioning, upgrade and Compass Runtime Agent reconnecting is uniquely identified by OperationID.
+- Runtime is uniquely identified by RuntimeID.
+- Before runtime is provisioned it must be registered in Director API. RuntimeID returned from Director should be used in Provisioner API.
```

Comment:
```suggestion
- Before you provision the Runtime, register it in Director API. Use RuntimeID returned from Director in Provisioner API.
```
Avoid using passive voice as it's more difficult to read for non-native speakers. 
You capitalize `Runtime` everywhere else - should it be capitalized here as well?